paper-server/src/main/java/io/papermc/paper/world/biome/PaperBiomeSpecialEffects.java

```diff
@@ -0,0 +1,93 @@
+package io.papermc.paper.world.biome;
+
+import io.papermc.paper.util.MCUtil;
+import io.papermc.paper.world.biome.effects.AdditionSound;
+import io.papermc.paper.world.biome.effects.MoodSound;
+import io.papermc.paper.world.biome.effects.MusicEntry;
+import io.papermc.paper.world.biome.effects.PaperAdditionSound;
+import io.papermc.paper.world.biome.effects.PaperMoodSound;
+import io.papermc.paper.world.biome.effects.PaperMusicEntry;
+import net.minecraft.core.Holder;
+import net.minecraft.sounds.Music;
+import net.minecraft.util.random.WeightedList;
+import net.minecraft.world.level.biome.Biome;
+import org.bukkit.Color;
+import org.bukkit.Location;
+import org.bukkit.Sound;
+import org.bukkit.craftbukkit.CraftSound;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+public class PaperBiomeSpecialEffects implements BiomeSpecialEffects {
+    private final Holder<Biome> biome;
+
+    public PaperBiomeSpecialEffects(Holder<Biome> biome) {
```

Comment:
It would make more sense to pass the internal equivalent to this class instead of just the biome, same for the other ones in this package